app/services/flightAPI.js

```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+
+const fetchFlightDetails = async (flightNumber) => {
+  try {
+    const response = await axios.get('https://667a89c2bd627f0dcc8f6b3b.mockapi.io/api/v1/flights', {
+      params: { flightNumber },
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.data.length) {
+      throw new Error('No flight details found');
+    }
+
+    return response.data.find(flight => flight.flightNumber === flightNumber);
```

Comment:
looks like `response.data` is already filtered by 
`params: { flightNumber }` at row #6